subprojects/packagefiles/catch2/src/catch2/meson.build

```diff
@@ -0,0 +1,338 @@
+  'matchers/catch_matchers_exception.cpp',
+  'matchers/catch_matchers_predicate.cpp',
+  'matchers/internal/catch_matchers_impl.cpp',
+  'catch_tag_alias_autoregistrar.cpp',
+  'internal/catch_decomposer.cpp',
+  'internal/catch_errno_guard.cpp',
+  'internal/catch_lazy_expr.cpp',
+  'internal/catch_leak_detector.cpp',
+  'internal/catch_message_info.cpp',
+  'internal/catch_polyfills.cpp',
+  'internal/catch_startup_exception_registry.cpp',
+  'internal/catch_uncaught_exceptions.cpp',
+  'interfaces/catch_interfaces_capture.cpp',
+  'interfaces/catch_interfaces_config.cpp',
+  'interfaces/catch_interfaces_exception.cpp',
+  'interfaces/catch_interfaces_registry_hub.cpp',
+  'interfaces/catch_interfaces_reporter_factory.cpp',
+  'interfaces/catch_interfaces_reporter_registry.cpp',
+  'interfaces/catch_interfaces_testcase.cpp',
+]
+
+reporter_headers = [
+  'reporters/catch_reporters_all.hpp',
+  'reporters/catch_reporter_automake.hpp',
+  'reporters/catch_reporter_common_base.hpp',
+  'reporters/catch_reporter_compact.hpp',
+  'reporters/catch_reporter_console.hpp',
+  'reporters/catch_reporter_cumulative_base.hpp',
+  'reporters/catch_reporter_event_listener.hpp',
+  'reporters/catch_reporter_helpers.hpp',
+  'reporters/catch_reporter_junit.hpp',
+  'reporters/catch_reporter_multi.hpp',
+  'reporters/catch_reporter_registrars.hpp',
+  'reporters/catch_reporter_sonarqube.hpp',
+  'reporters/catch_reporter_streaming_base.hpp',
+  'reporters/catch_reporter_tap.hpp',
+  'reporters/catch_reporter_teamcity.hpp',
+  'reporters/catch_reporter_xml.hpp'
+]
+
+reporter_sources = [
+  'reporters/catch_reporter_automake.cpp',
+  'reporters/catch_reporter_common_base.cpp',
+  'reporters/catch_reporter_compact.cpp',
+  'reporters/catch_reporter_console.cpp',
+  'reporters/catch_reporter_cumulative_base.cpp',
+  'reporters/catch_reporter_event_listener.cpp',
+  'reporters/catch_reporter_helpers.cpp',
+  'reporters/catch_reporter_junit.cpp',
+  'reporters/catch_reporter_multi.cpp',
+  'reporters/catch_reporter_registrars.cpp',
+  'reporters/catch_reporter_sonarqube.cpp',
+  'reporters/catch_reporter_streaming_base.cpp',
+  'reporters/catch_reporter_tap.cpp',
+  'reporters/catch_reporter_teamcity.cpp',
+  'reporters/catch_reporter_xml.cpp',
+]
+
+headers = benchmark_headers + internal_headers + reporter_headers
+sources = benchmark_sources + internal_sources + reporter_sources
+
+# The headers must be installed with their full paths, which meson
+# v0.63 supports using `preserve_path` for `install_headers`. We'd
+# like to be compatible with Debian 11 (current stable) which has
+# meson v0.56.2. Instead, let's use the technique from
+# https://github.com/mesonbuild/meson/issues/14 with some tweaks to
+# make it compatible with newer meson.
+include_subdir = 'catch2'
+foreach file : headers
+    file_path = file.split('/')
+
+    folder = ''
+    foreach path : file_path
+        if path != file_path[-1]
+            folder = folder / path
```

Comment:
This is currently the used feature with the highest minimum version of Meson, at a pretty backwards compatible 0.49.0 -- great stuff.

Can we assert that in code by adding `meson_version: '>=0.49.0'` in the `project()` function kwargs?